packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -0,0 +1,310 @@
+  | "myContent"
+  | "myGroup"
+  | "favorites"
+  | "organization"
+  | "world";
+
+/**
+ * This is used to determin what IHubCollection definition JSON object
+ * can be created. We use HubFamily here to define most of the collections
+ * to ensure consistency
+ */
+export type WellKnownCollection =
+  | Exclude<HubFamily, "app" | "map">
+  | "appAndMap"
+  | "solution";
+
+/**
+ * A list of optional arguments to pass into getWellKnownCatalog
+ */
+interface IGetWellKnownCatalogOptions {
+  user?: IUser;
+  collectionNames?: WellKnownCollection[];
+}
+
+/**
+ * Check if i18nScope is defined, if so add a "." at the end
+ * @param i18nScope
+ * @returns i18nScope with a "." at the end if it is defined so it stays
+ * as a valid i18nScope
+ * e.g. "project.edit.catalog" vs "project.editcatalog"
+ */
+function validateI18nScope(i18nScope: string): string {
+  return i18nScope ? `${i18nScope}.` : "";
+}
+
+/** Get a single IHubCatalog definition JSON object based on the
+ * name and entity type requested
+ * @param i18nScope Translation scope to be interpolated into the catalog
+ * @param name Name of the catalog requested
+ * @param entityType
+ * @param user Owner of the entity, optional but required for certain catalogs
+ * @param collectionNames A list of collection names, optional, if passed in,
+ * only those collections in the catalog will be returned
+ */
+export function getWellKnownCatalog(
+  i18nScope: string,
+  name: WellKnownCatalog,
+  entityType: EntityType,
+  options?: IGetWellKnownCatalogOptions
+): IHubCatalog {
+  switch (entityType) {
+    case "item":
+      return getWellknownItemCatalog(i18nScope, name, options);
+    /* Add other entity handlers here, e.g. getWellknownEventCatalog */
+    default:
+      throw new Error(`Wellknown catalog not implemented for "${entityType}"`);
+  }
+}
+
+/**
+ * Build an IHubCatalog definition JSON object based on the
+ * catalog name, predicate and collections we want to use for each catalog
+ * @param i18nScope
+ * @param name Catalog name
+ * @param predicates Predicates for the catalog
+ * @param collections Collections to include for the catalog
+ * @returns An IHubCatalog definition JSON object
+ */
+function buildCatalog(
+  i18nScope: string,
+  name: WellKnownCatalog,
+  predicates: any[],
+  collections: IHubCollection[]
+): IHubCatalog {
+  return {
+    schemaVersion: 1,
+    title: `{{${i18nScope}catalog.${name}:translate}}`,
+    scopes: {
+      item: {
+        targetEntity: "item" as EntityType,
+        filters: [{ predicates }],
+      },
+    },
+    collections,
+  };
+}
+
+/**
+ * Validate the arguments based on its type
+ * @param type type of arg to validate
+ * @param name name of the catalog
+ * @param args arguments to validate
+ */
+function validateArgs(type: string, name: WellKnownCatalog, args: any): any {
+  switch (type) {
+    case "user":
+      if (!args?.user) {
+        throw new Error(`User needed to get "${name}" catalog`);
+      }
+  }
+}
```

Comment:
@sonofflynn89 took your advice, I'm not sure if I overcomplicated the logic though, we could make it simpler by just checking the user, but I thought this is more expandable.

Reply:
To me this is confusing... the args you are validating are typed as `IGetWellKnownCatalogOptions` so I think you should take that in vs `any`... I'm also not wild about having a "validation" fn that throws... I'd rather this returns a structure like `{valid: false, message: "User needed to get "user" catalog"}` and then have the caller throw or decide how to handle this. It will also make debugging easier b/c the call stack will be less confusing.

Reply:
Agreed about keeping the validation function pure, since we are only checking if the passed options have certain props we need, it's really unnecessary to have the validation function anymore, I'm gonna move the check and throw back to the main function for now, if we have more conditions to check in the future, we will come back and re-evaluate. Since I've done some other cleanups, the main function is less cumbersome now even with the 'if's.

Reply:
Or, have a simpler function called `checkUser` just for checking the user and throw, to keep it DIY